hyperspyui/plugins/alignzlp.py

```diff
@@ -24,4 +31,19 @@ def create_toolbars(self):
     def default(self):
         ui = self.ui
         s = ui.get_selected_signal()
-        s.align_zero_loss_peak()
+        logger.debug('Align zero loss peak of ' + str(s))
+        title = s.metadata.General.title
+
+        signal_list = [
+            sig for sig in ui.signals if sig.signal is not s and type(sig) == hyperspy.signals.EELSSpectrum]
+        picker = SignalList(items=signal_list, parent=ui, multiselect=True)
```

Comment:
I would suggest the following change to make it more readable and display the dialog only when appropriate:
```python
        also_align = []
        if len(signal_list) > 0:
            picker = SignalList(items=signal_list, parent=ui, multiselect=True)
            diag = ui.show_okcancel_dialog("Select signals to align with {}".format(
                title), picker, modal=True)
            if diag.result() != QDialog.Accepted:
                return
            also_align = [sig.signal for sig in picker.get_selected()]
        logger.debug(
            'Also aligning the following signals\n' + str(also_align))
        s.align_zero_loss_peak(also_align=also_align)
```